File: CS/RadialMenuExample/TextBoxService.cs
using DevExpress.Mvvm.UI;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Controls;

namespace RadialMenuExample
{
    public interface ITextBoxService
    {
        void ClearSelection();
        bool CanClearSelection();
    }

    //This service is used for the Clear command. To learn more about custom services, refer to https://documentation.devexpress.com/#WPF/CustomDocument16920
    public class TextBoxService : ServiceBase, ITextBoxService
    {
        TextBox MyTextBox { get { return AssociatedObject as TextBox; } }
        public void ClearSelection()
        {
            MyTextBox.SelectionLength = 0;
        }
        public bool CanClearSelection()
        {
            return MyTextBox.SelectionLength > 0;
        }
    }
}

File: VB/RadialMenuExample/TextBoxService.vb
Imports DevExpress.Mvvm.UI
Imports System.Windows.Controls

Namespace RadialMenuExample

    Public Interface ITextBoxService

        Sub ClearSelection()

        Function CanClearSelection() As Boolean

    End Interface

    'This service is used for the Clear command. To learn more about custom services, refer to https://documentation.devexpress.com/#WPF/CustomDocument16920
    Public Class TextBoxService
        Inherits ServiceBase
        Implements ITextBoxService

        Private ReadOnly Property MyTextBox As TextBox
            Get
                Return TryCast(AssociatedObject, TextBox)
            End Get
        End Property

        Public Sub ClearSelection() Implements ITextBoxService.ClearSelection
            MyTextBox.SelectionLength = 0
        End Sub

        Public Function CanClearSelection() As Boolean Implements ITextBoxService.CanClearSelection
            Return MyTextBox.SelectionLength > 0
        End Function
    End Class
End Namespace
